compiler/src/iree/compiler/Codegen/LLVMGPU/test/configure_tensor_layout.mlir

```diff
@@ -266,3 +266,27 @@ func.func @gather_like(%base : tensor<16384x16x32x128xf16>,
 // CHECK-LABEL: func.func @gather_like
 // CHECK: %[[OUT:.+]] = linalg.generic
 // CHECK: to_layout %[[OUT]] to layout(#[[$LAYOUT]])
+
+// -----
+
+#translation = #iree_codegen.translation_info<pipeline = LLVMGPUVectorDistribute
+                                              workgroup_size = [64, 1, 1]
+                                              subgroup_size = 64>
+
+func.func @dynamic_infer_sizes(%in : tensor<4x32x?x128xf16>) -> tensor<1x1x?x128xf16> attributes { translation_info = #translation } {
+  %c0 = arith.constant 0 : index
+  %c2 = arith.constant 2 : index
+  %d2 = tensor.dim %in, %c2 : tensor<4x32x?x128xf16>
+  %45 = affine.min affine_map<(d0)[s0] -> (-d0 + s0, 1024)>(%c0)[%d2]
+  %extracted_slice_5 = tensor.extract_slice %in[%c0, %c0, %c0, 0] [1, 1, %45, 128] [1, 1, 1, 1] : tensor<4x32x?x128xf16> to tensor<1x1x?x128xf16>
+  %49 = tensor.empty(%45) : tensor<1x1x?x128xf16>
+  %50 = linalg.copy {lowering_config = #iree_gpu.derived_thread_config} ins(%extracted_slice_5 : tensor<1x1x?x128xf16>) outs(%49 : tensor<1x1x?x128xf16>) -> tensor<1x1x?x128xf16>
```

Comment:
Interesting... this is inferring the size of the dynamic dimension to be 1024. That's wrong right? The minimum size of the dynamic dimension is 1024, not the maximum size?

Reply:
err.... no ? 
when you do a y = min(x, 1024), it guarantees max(y) = 1024 ?

Reply:
I don't know why I hallucinated here. You are right. Sorry :P